### DISCUSSION TITLE
[Bug]: Radio button in Controller

### DISCUSSION THREAD

**Discussion by taily-khucnaykhongquantrong** [react-hook-form/react-hook-form#6580] [Q&A] [answered]
### Version Number

7.15.3

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-get-started-forked-l5k1e?file=/src/index.js

### Steps to reproduce

1. Click on Controller radio button to change value to "1".
2. Click submit.
3. Click on radio button which value is "1" again.
4. Click submit.
5. Click submit again.

### Expected behaviour

- On first submit, form values should be `{ test_controller: "1" }`.
- On third submit, form values should be `{ test_controller: "1" }`.
- And default checked doesn't work on Controller.
- If you switch the default value of Controller to "1", it will work fine.
- At the moment, I have to switch to controlled component as a workaround to make it work and destructure `formState` even though I'm not using it (picture 2).
![image](https://user-images.githubusercontent.com/9252925/133878520-5b28ab6b-7649-41ec-944e-38de3c5ddca6.png)
![image](https://user-images.githubusercontent.com/9252925/133878542-be7952a4-d2f7-45d7-a3f5-9d4e206a5e21.png)

### What browsers are you seeing the problem on?

Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
This is not an issue to do with Controller, Controller only collects and validates the data. 

you should build a controlled component which host those radio input with useState and have controller wrapped around it.

eg:

```tsx
const Radios = ({value, onChange, onBlur}) => {
  const [value, setValue ] = React.useState(value)
  return [options].map(() => <input onChange={() => {
    setValue('...update your local component');
    onChange('...send value to react hook form');
  }} />)
}

<Controller render={({ field }) => <Radios {...field} />} />
````

an example with MUI: https://codesandbox.io/s/react-hook-form-v7-controller-5h1q5?file=/src/Mui.js
<img width="747" alt="Screen Shot 2021-09-18 at 4 46 05 pm" src="https://user-images.githubusercontent.com/10513364/133879239-45d1413f-599d-4e76-8ead-5a989128013e.png">

**Reply by taily-khucnaykhongquantrong** [reply]
So Controller is used for only controlled component?

**Reply by bluebill1049** [reply]
yes, it's used when you want an external controlled component (eg: other libs components). It only collects value and validates.
